utils.groovy

```diff
@@ -835,16 +835,24 @@ def build_remote_image(arches, commit, url, repo, tag, registry_secret_path, sec
         [arch, {
             pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
                 def final_args = build_args + ["--arch", arch, "--tag", "${tag}-${arch}"] + extra_args
+                if (store_digest) {
+                     final_args += ["--write-digest-to-file", "${tag}-${arch}"]
```

Comment:
Hmm, a bit odd that we "leak" the concept of that file across two separate functions.

How about instead, we have `build_remote_image` return the list of digests (and it can delete those files as well). And then in `build_and_push_image`, we just take that list and pass it to `push_manifest`?

I think if we do that, let's just have the logic always do it. I.e. let's not have any `store_digest` parameter on any of the functions.